core/src/main/java/org/apache/cxf/endpoint/ClientImpl.java

```diff
@@ -539,7 +539,7 @@ public void onMessage(Message message) {
             return processResult(message, exchange, oi, resContext);
         } finally {
             //ensure ResponseContext has HTTP RESPONSE CODE
-            if (null != exchange) {
+            if (null != exchange && null != responseContext) {
```

Comment:
Thank you @ctabin , could you please guard the `setResponseContext(resContext);` instead?

```
if (null != responseContext) {
    setResponseContext(resContext);
}
```

The reason is that is should be fine to fill up the `resContext` (which is attached to the current message). Thank you

Reply:
@reta Done :thumbsup: